applications/GeoMechanicsApplication/custom_utilities/transport_equation_utilities.hpp

```diff
@@ -0,0 +1,68 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Ignasi de Pouplana,
+//                   Vahid Galavi
+//                   Gennady Markelov
+//
+
+#pragma once
+
+// Project includes
+
+// Application includes
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class GeoTransportEquationUtilities
+{
+public:
+    using VectorType = Vector;
+    using MatrixType = Matrix;
+
+    //----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    template <unsigned int TDim, unsigned int TNumNodes>
+    static inline void CalculatePermeabilityMatrixH(BoundedMatrix<double, TNumNodes, TNumNodes>& rPMatrix,
+                                                    const Matrix& GradNpT,
+                                                    double        DynamicViscosityInverse,
+                                                    const BoundedMatrix<double, TDim, TDim>& PermeabilityMatrix,
+                                                    double IntegrationCoefficient)
+    {
+        noalias(rPMatrix) = DynamicViscosityInverse *
+                            prod(Matrix(prod(GradNpT, PermeabilityMatrix)), trans(GradNpT)) * IntegrationCoefficient;
+    }
+
+    static inline Matrix CalculatePermeabilityMatrixH(const Matrix& GradNpT,
```

Comment:
As far as understand only Integration coefficient blocks it because it is calculated for each point. Of course, it can be moved  separately and kept as a matrix. Looking forward to @WPK4FEM and @avdg81 suggestion.

Reply:
Independently of @rfaasse I had the same thoughts about letting this function return the matrix for an element, rather than an integration point contribution. I would suggest to make that change in this PR too. Let's sit together and explore how that can be done.

Reply:
Yesterday Anne and me had a conversation on this. The decision is to keep the current approach and to make changes later.